### PR TITLE
Guide for using proxy with docker in pipeline

### PR DESCRIPTION
**Pull request by abudevich**

### CODE REVIEW

File: docs/modules/guides/pages/pipeline-with-proxy.adoc [hunk 1]
```diff
@@ -0,0 +1,85 @@
+= Example of pipeline script with starting VIVIDUS proxy in DOCKER container
```
Comment (ikalinin1) [inline docs/modules/guides/pages/pipeline-with-proxy.adoc:1]: Do you mind to put some comments about the sections of the file?

Here is an example 

https://github.com/vividus-framework/vividus/blame/03bfc123f0cd96b32dbd3b6b7f94bcaca3be283a/docs/modules/ROOT/pages/tests-configuration.adoc#L106-L132
image

File: docs/modules/guides/pages/pipeline-with-proxy.adoc [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+```bash
```
Comment (valfirst) [inline docs/modules/guides/pages/pipeline-with-proxy.adoc:3]:
```suggestion
```groovy
```

File: docs/modules/guides/pages/pipeline-with-proxy.adoc [hunk 3]
```diff
@@ -0,0 +1,90 @@
+= Example of pipeline script with starting VIVIDUS proxy in DOCKER container
```
Comment (valfirst) [inline docs/modules/guides/pages/pipeline-with-proxy.adoc:1]:
```suggestion
= Jenkins pipeline using browser proxy in Docker container
```

File: docs/modules/guides/nav.adoc [hunk 4]
```diff
@@ -1,3 +1,5 @@
 .User Guides
 * Data Testing
 ** xref:execute-sql-against-csv.adoc[Execute SQL against CSV]
+* CI Integration
+** xref:pipeline-with-proxy.adoc[Pipeline with proxy]
```
Comment (valfirst) [inline docs/modules/guides/nav.adoc:5]:
```suggestion
** xref:pipeline-with-proxy.adoc[Jenkins pipeline with browser proxy]
```

File: docs/modules/guides/pages/pipeline-with-proxy.adoc [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        }
+    }
+
+    stage('Load Source') {
+        steps {
+            checkout([$class: 'GitSCM',
+                branches: [[name: "master"]],
+                doGenerateSubmoduleConfigurations: false,
+                extensions: [
+                    [$class: 'RelativeTargetDirectory', relativeTargetDir: "${projectName}"],
```
Comment (valfirst) [inline docs/modules/guides/pages/pipeline-with-proxy.adoc:35]: please add variable with some dummy value for `${projectName}`

**Review by valfirst — CHANGES_REQUESTED** [review]

File: docs/modules/guides/pages/pipeline-with-proxy.adoc [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+                    [$class: 'SubmoduleOption',
+                    disableSubmodules: false,
+                    parentCredentials: true,
+                    recursiveSubmodules: true,
+                    reference: '',
+                    trackingSubmodules: false]
+                ],
+                submoduleCfg: [],
+                userRemoteConfigs: [[credentialsId: "${scmCredentialId}", url: "${gitUrl}"]]])
```
Comment (uarlouski) [inline docs/modules/guides/pages/pipeline-with-proxy.adoc:44]: pass just variable here and in other places

File: docs/modules/guides/pages/pipeline-with-proxy.adoc [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+                    [$class: 'RelativeTargetDirectory', relativeTargetDir: projectName],
+                    [$class: 'SubmoduleOption',
+                    disableSubmodules: false,
+                    parentCredentials: true,
+                    recursiveSubmodules: true,
+                    reference: '',
+                    trackingSubmodules: false]
+                ],
+                submoduleCfg: [],
+                userRemoteConfigs: [[credentialsId: scmCredentialId, url: gitUrl]]])
```
Comment (valfirst) [inline docs/modules/guides/pages/pipeline-with-proxy.adoc:46]: how does it work? I am curious about `name: master`, `relativeTargetDir: projectName`, `credentialsId: scmCredentialId, url: gitUrl`

**Review by uarlouski — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**